ac/ac-classifier/src/Classifier.jsx

```diff
@@ -124,33 +87,44 @@ class Runner extends Component<Object, RunnerStateT> {
   }
 
   render() {
-    const { activityData, data, dataFn, setObjectKey } = this.props;
+    const { activityData, objectKey, data, dataFn, setObjectKey } = this.props;
     const { objectKeyPlus, objects, categories } = this.state;
     return (
       <Main>
         <h2>{activityData.config.title}</h2>
-        {objectKeyPlus ? (
-          <FlexDiv>
-            <ObjectPanel obj={data[objectKeyPlus]} small={false} />
-            {categories.length > 0 && (
+        <FlexDiv>
+          {objectKey && (
+            <this.props.LearningItem id={objectKey} type="history" />
+          )}
+          {categories.length > 0 && (
+            <div style={{ position: 'absolute', right: '20px' }}>
               <ShortcutPanel
                 {...{
                   categories,
-                  dataFn,
+                  dataFn: this.props.ourDataFn,
                   data,
                   assignCategory: this.assignCategory,
-                  objectKey: objectKeyPlus
+                  objectKey: objectKey
                 }}
               />
-            )}
-          </FlexDiv>
-        ) : (
-          <h1>Waiting for objects to classify</h1>
-        )}
-        <ObjectList {...{ objects, objectKey: objectKeyPlus, setObjectKey }} />
+            </div>
+          )}
+        </FlexDiv>
+        <ObjectList
+          {...{
+            LearningItem: this.props.LearningItem,
+            objects: data,
+            objectKey: objectKey,
+            setObjectKey,
+            setDataFn: this.props.setDataFn
+          }}
+        />
       </Main>
     );
   }
 }
 
-export default withState('objectKey', 'setObjectKey', null)(Runner);
+export default compose(
+  withState('ourDataFn', 'setDataFn', undefined),
```

Comment:
Should rename to `learningItemDataFn` or `liDataFn`, otherwise I will call all my components `MyComponent`

Reply:
I don't really understand what is happening here actually.

Comparing that code with the code also found in ac-brainstorm. The `dataFn` props that we have when we use the `render` props of the `LearningItem` component, makes modifications on the metadata. Here you have implemented this in the thumbnail list to access the `dataFn` of teh LI we are working on, but couldn't you have accessed it where you display the LI in mode history?